src/main/java/executor/service/DIFactory/ObjectFactoryImpl.java

```diff
@@ -0,0 +1,81 @@
+package executor.service.DIFactory;
+
+import org.reflections.Reflections;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class ObjectFactoryImpl implements ObjectFactory {
+    private static final Singleton INSTANCE = Singleton.INSTANCE;
+
+    private volatile static ObjectFactoryImpl instance = null;
+    private ObjectFactoryImpl() {}
+    public static ObjectFactoryImpl getInstance() {
+        if (instance == null) {
+            synchronized(ObjectFactoryImpl.class) {
+                if (instance == null) {
+                    instance = new ObjectFactoryImpl();
+                }
+            }
+        }
```

Comment:
There are no sense to make like this, ENUM initialize on bootstrap, and it always single.

Reply:
So ObjectFactoryImpl doesn't need to be Singleton?